sympy/integrals/intpoly.py

```diff
@@ -294,21 +303,44 @@ def main_integrate(expr, facets, hp_params, max_degree=None):
                                 (b / norm(a)) / (dim_length + degree)
         return result
     else:
-        polynomials = decompose(expr)
-        for deg in polynomials:
-            poly_contribute = S.Zero
-            facet_count = 0
-            for hp in hp_params:
-                value_over_boundary = integration_reduction(facets,
-                                                            facet_count,
-                                                            hp[0], hp[1],
-                                                            polynomials[deg],
-                                                            dims, deg)
-                poly_contribute += value_over_boundary * (hp[1] / norm(hp[0]))
-                facet_count += 1
-            poly_contribute /= (dim_length + deg)
-            integral_value += poly_contribute
-    return integral_value
+        if not isinstance(expr,list):
+            polynomials = decompose(expr)
```

Comment:
Try to factor out this code into a (private) function and use here and for the inner loop below.

Reply:
Yeah I have done this , hope that the name used is suitable !

Reply:
Looks good! Thanks! (Good that you used a leading underscore.)